src/utils/simplifier.js

```diff
@@ -308,6 +308,23 @@ function simplifyEquality(realm: Realm, equality: AbstractValue): Value {
       if (simplified(sxx) || simplified(sxy))
         return AbstractValue.createFromConditionalOp(realm, cond, sxx, sxy, equality.expressionLocation, true);
     }
+  } else {
+    // !((c ? x : y) === x) && x !== y <=> !c
```

Comment:
There is a mismatch between the letters in the comment, and the letters in the implementation.

Reply:
The actual implementation only does this for concrete values, while the rule makes a more general statement. A comment would be good if this is intentional or just a practical limitation.

Reply:
This mismatch is intentional because the two cases below are specializations of the general idea of the comment.